File: problem6.cs
using System;

namespace problem6
{
    class Program
    {
        static void Main(string[] args)
        {
            /*
             Project Euler Problem 6

            EN:The sum of the squares of the first ten natural numbers is 1^2 + 2^2 +...+ 10^2 = 385,
            The square of the sum of the first ten natural numbers is (1+2+...+10)^2 = 55^2 = 3025,
            Hence the difference between the sum of the squares of the first ten natural numbers and the square of the sum is 3025 - 385 = 2640.
            Find the difference between the sum of the squares of the first one hundred natural numbers and the square of the sum.

            TR:İlk on doğal sayının kareleri toplamı 1^2 + 2^2 +...+ 10^2 = 385'dir,
            İlk on doğal sayının toplamının karesi (1+2+...+10)^2 = 55^2 = 3025'dir,
            İlk on doğal sayının toplamlarının karesi ve karelerinin toplamının farkı 3025-385'den 2640'dır.
            İlk yüz doğal sayının toplamlarının karesinin ve karelerinin toplamının farkını bulunuz.
             */
            int toplamlarinKaresi,karelerininToplami=0, sayilarinToplami=0, fark;
            for (int i = 1; i <= 100; i++)
            {
                karelerininToplami += i * i;
                sayilarinToplami += i;
            }
            toplamlarinKaresi = sayilarinToplami * sayilarinToplami;
            fark = toplamlarinKaresi - karelerininToplami;
            Console.WriteLine("İlk 100 doğal sayının toplamlarının karesinin farkı ile karelerinin toplamının farkı: {0} sayısıdır.",fark);
        }
    }
}

File: problem6.py
"""
Project Euler Problem 6

EN:The sum of the squares of the first ten natural numbers is 385,
The square of the sum of the first ten natural numbers is 3025 ,
Hence the difference between the sum of the squares of the first ten natural numbers 
and the square of the sum is 2640 .
Find the difference between the sum of the squares of the first one hundred
 natural numbers and the square of the sum.
 
TUR: İlk 10 doğal sayının karelerinin toplamı 385'tir.
İlk 10 doğal sayının toplamının karesi ise 3025'tir.
Bu nedenle ilk 10 doğal sayının toplamının karesi ile ilk 10 doğal sayının karesinin toplamının 
arasındaki fark 2640'tır.

İlk yüz doğal sayının karelerinin toplamı ile ilk yüz sayının toplamının karesinin arasındaki
farkı bulun.
"""

first_100_square_sum = sum([i**2 for i in range(1,101)])
first_100_sum_square = (sum([i for i in range(1,101)]) ** 2)

print(abs(first_100_sum_square-first_100_square_sum))

